LBHD, Directional Armor v1.2.lsl

```diff
@@ -0,0 +1,206 @@
+            //Now we use the Z-Axis to calculate the horizonal directions.
+            //Note that vertical direction isn't factored, only the horizonal angle. This the vehicle is sliced up like a pie and damage will be based on how far and which direction from the center the projectile strikes when it hits the top or bottom.
+            if(trotvec.z>-20.0&&trotvec.z<20.0)//Front
+                return llFloor((dmg*front)*mod);
+            else if(trotvec.z<-160.0||trotvec.z>160.0)//Back
+                return llFloor((dmg*back)*mod);
+            else //If it didn't hit any previous angles, the only thing left to hit is the sides.
+                return llFloor((dmg*side)*mod);
+        }
+    }
+    else return 0;//If a no vector is returned, do not process damage.
+}
+//Damage Processor
+damage(integer amt, key id,vector pos, vector tpos)
+{
+    /*if(amt<0)//Allows the object to be healed/repaired
+    {
+        if(llGetTime()>1.0)//Optional healing cooldown
+        {
+            if(amt>hp*0.1)amt=hp*0.1;//Optional healing cap
+            hp-=amt;
+            llResetTime();
+        }
+        //Be sure to update the listen event code block to allow negative damage values through.
+    }*/
+    /*else if(amt<6)return; //Blocks micro-LBA
+    else*/
+    {
+        integer directional_amt=lbapos(amt,pos,tpos);
+        if(directional_amt)hp-=directional_amt;
+        else
+        {
+            //llRegionSayTo(llGetOwnerKey(id),0,"/me Armor deflected the damage!");//cheeki breeki
+            return;
+        }
+        //llSay(0,"/me took "+(string)amt+" LBA/LBB damage!");//Used to debug output.
+    }
+    if(hp<1)die();
+    else
+    {
+        if(hp>mhp)hp=mhp;//Used to prevent overhealing
+        update();
+    }
+}
+//Line-of-Sight Check
+integer los(vector start, vector target)
+{
+    list ray=llCastRay(target,start,[RC_REJECT_TYPES,RC_REJECT_AGENTS,RC_DATA_FLAGS,RC_GET_ROOT_KEY,RC_MAX_HITS,1]);
+    key hit=llList2Key(ray,0);//Debug
+    if(llKey2Name(hit))
+    {
+        //llSay(0,llKey2Name(hit));//Debug
+        if(hit==me)return 1;
+        else return 0;//Object in way
+    }
+    else
+    {
+        if(llList2Vector(ray,1)==ZERO_VECTOR)return 1;
+        else return 0;//Land in way
+    }
+}
+update()//SetText
+{
+    llSetLinkPrimitiveParamsFast(-4,[PRIM_TEXT,"[LBHD]\n "+(string)hp+" / "+(string)mhp+" HP",<0.0,0.75,1.0>,1.0,
+        PRIM_DESC,"LBA.v.LBHD"+(string)hp+","+(string)mhp+","+(string)atcap+",999"+
+        //In order: Current HP, Max HP, Max AT accepted, Max healing accepted (Not implemented)
+        ",F-"+llGetSubString((string)front,0,2)+//Frontal modifier
+        ",S-"+llGetSubString((string)side,0,2)+//Side modifier
+        ",R-"+llGetSubString((string)back,0,2)+//Rear modifier
+        ",T-"+llGetSubString((string)top,0,2)+//Top modifier
+        ",B-"+llGetSubString((string)bottom,0,2)+//Bottom modifier
+        ",M-"+llGetSubString((string)middle,0,2)]);//Middle Modifier
```

Comment:
This section adds a lengthy string to the object description. Consider combining all of the parameters into one instead of adding one per "modifier", or removing them entirely. There's no tangible method for scripts to use them on the fly, and scripts that attempt to are unlikely to accurately predict damage after modifiers due to the nature of SL physics and unpredictable script delays.

Reply:
That is mainly there so people can inspect them and make sure the vehicle/object is compliant with rules regarding damage modifiers.